### DISCUSSION TITLE
Thanos Qurier: Is there any http or grpc client for querier ?

### DISCUSSION THREAD

**Discussion by imaskm** [thanos-io/thanos#5141] [Questions & Answers]
Hi Guys, we have a use case where we want to fetch metrics from Thanos Querier, is there any  HTTP client/sdk or even grpc client for this use case?

**Comment by LukaszWasko** [comment]
maybe this?
./grpcurl -plaintext 0.0.0.0:10904 list grpc.health.v1.Health
more: https://github.com/fullstorydev/grpcurl

**Comment by yeya24** [comment]
@imaskm Do you want to fetch Thanos Querier's metrics or use its Query API? If you want to fetch its own metrics then you can just use Prometheus. If the latter then you can use Prometheus HTTP API.

**Reply by imaskm** [reply]
@yeya24  I want to call the thanos query API, I know they are compatible with Prometheus and I am using prometheus client now but I was just wondering if there is any client or SDK  which is specifically implemented to access Querier API

**Reply by yeya24** [reply]
Unfortunately there is no such thing right now. Usually prometheus client is sufficient but it lacks some Thanos specific parameters like resolution, deduplication, etc. Ideally you can implement it your own easily.
